crates/ruff_linter/src/rules/isort/sorting.rs

```diff
@@ -1,171 +1,137 @@
 /// See: <https://github.com/PyCQA/isort/blob/12cc5fbd67eebf92eb2213b03c07b138ae1fb448/isort/sorting.py#L13>
+use natord;
 use std::cmp::Ordering;
-use std::collections::BTreeSet;
 
 use ruff_python_stdlib::str;
 
 use super::settings::{RelativeImportsOrder, Settings};
-use super::types::EitherImport::{Import, ImportFrom};
-use super::types::{AliasData, EitherImport, ImportFromData, Importable};
-
-#[derive(PartialOrd, Ord, PartialEq, Eq, Copy, Clone)]
-pub(crate) enum Prefix {
-    Constants,
-    Classes,
-    Variables,
+
+#[derive(Debug, PartialOrd, Ord, PartialEq, Eq, Copy, Clone)]
+pub(crate) enum MemberType {
+    Constant,
+    Class,
+    Variable,
 }
 
-fn prefix(name: &str, settings: &Settings) -> Prefix {
+fn member_type(name: &str, settings: &Settings) -> MemberType {
     if settings.constants.contains(name) {
         // Ex) `CONSTANT`
-        Prefix::Constants
+        MemberType::Constant
     } else if settings.classes.contains(name) {
         // Ex) `CLASS`
-        Prefix::Classes
+        MemberType::Class
     } else if settings.variables.contains(name) {
         // Ex) `variable`
-        Prefix::Variables
+        MemberType::Variable
     } else if name.len() > 1 && str::is_cased_uppercase(name) {
         // Ex) `CONSTANT`
-        Prefix::Constants
+        MemberType::Constant
     } else if name.chars().next().is_some_and(char::is_uppercase) {
         // Ex) `Class`
-        Prefix::Classes
+        MemberType::Class
     } else {
         // Ex) `variable`
-        Prefix::Variables
+        MemberType::Variable
     }
 }
 
-/// Compare two module names' by their `force-to-top`ness.
-fn cmp_force_to_top(name1: &str, name2: &str, force_to_top: &BTreeSet<String>) -> Ordering {
-    let force_to_top1 = force_to_top.contains(name1);
-    let force_to_top2 = force_to_top.contains(name2);
-    force_to_top1.cmp(&force_to_top2).reverse()
-}
+#[derive(Eq, PartialEq, Debug)]
+pub(crate) struct NatOrdString(String);
 
-/// Compare two top-level modules.
-pub(crate) fn cmp_modules(alias1: &AliasData, alias2: &AliasData, settings: &Settings) -> Ordering {
-    cmp_force_to_top(alias1.name, alias2.name, &settings.force_to_top)
-        .then_with(|| {
-            if settings.case_sensitive {
-                natord::compare(alias1.name, alias2.name)
-            } else {
-                natord::compare_ignore_case(alias1.name, alias2.name)
-                    .then_with(|| natord::compare(alias1.name, alias2.name))
-            }
```

Comment:
Is this logic still being captured in the refactor?

Reply:
It should be because both `module_key` and `member_key` return a tuple with first `maybe_lower_case_name` (which is `Some(name.to_lowercase())` if `settings.case_sensitive` is false, otherwise `None`) and then the module/member name.

So if `settings.case_sensitive` is true, then we just compare the names using `natord::compare`, and if it's false we first compare the lowercase names with `natord::compare`, and then the actual names.

I guess the question is, is comparing two strings that have been turned into lowercase with `natord::compare` the same as comparing them directly with `natord::compare_ignore_case`? I looked at the [source code](https://docs.rs/natord/latest/src/natord/lib.rs.html#142-153) for `natord::compare_ignore_case` and they're just calling `to_lowercase` on the `char`s individually, but apart from that it seems like what I did should be equivalent.

Reply:
I think it is slightly different, because they also map integers to avoid sorting them lexicographically, right? For example, to avoid putting `module10` before `module2`.

Reply:
Oh I thought you were talking about the fact that there are two function calls (one case sensitive and one case insensitive).

The correct natrural ordering (typically `module2` comes before `module10`) is indeed correctly captured in my refactor because I'm relying on `natord::compare` to implement the `Ord` trait for my types. Was that the question?